Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs

```diff
@@ -90,20 +79,18 @@ public async Task<Calendar> GetCalendarByIdAsync(string externalId)
         /// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
         public async Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
         {
-            await RequestCalendarAccess().ConfigureAwait(false);
+            _ = calendar.ExternalID ?? throw new ArgumentNullException("ExternalID cannot be null");
 
-            var deviceCalendar = _eventStore.GetCalendar(calendar.ExternalID);
+            await RequestCalendarAccess().ConfigureAwait(false);
 
-            if (deviceCalendar == null)
-            {
-                throw new ArgumentException("Specified calendar not found on device");
-            }
+            var deviceCalendar = _eventStore.GetCalendar(calendar.ExternalID)
+                ?? throw new ArgumentException("Specified calendar not found on device");
 
             var query = _eventStore.PredicateForEvents(start.ToNSDate(), end.ToNSDate(), new EKCalendar[] { deviceCalendar });
             var events = await Task.Run(() =>
             {
                 var iosEvents = _eventStore.EventsMatching(query);
-                return iosEvents == null ? new List<CalendarEvent>() : iosEvents.Select(e => e.ToCalendarEvent()).ToList();
```

Comment:
This change is not needed, but `EventsMatching` cannot return `null` according to it's attributes.
If this is a mistake, I'll revert it.